site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,694 @@
+// Continue reading to find out how to configure the instance.
+const i18n = new I18n({
+  defaultLocale: "en", // see below for more information
+
+  // Load all translation files from locales/.
+  directory: "locales",
+});
+
+// Finally, register the i18n instance in the bot,
+// so the messages get translated on their way!
+bot.use(i18n.middleware());
+
+// Everything is set up now.
+// You can access translations with `t` or `translate`.
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("start-msg"));
+});
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
 
 ```js
-import { Bot, session } from "grammy";
-import { I18n } from "@grammyjs/i18n";
+const { Bot } = require("grammy");
+const { I18n } = require("@grammyjs/i18n");
+
+// Create a bot as you normally would.
+const bot = new Bot(""); // <-- put your bot token here (https://t.me/BotFather)
+
+// Create an `I18n` instance.
+// Continue reading to find out how to configure the instance.
+const i18n = new I18n({
+  defaultLocale: "en", // see below for more information
+
+  // Load all translation files from locales/.
+  directory: "locales",
+});
 
+// Finally, register the i18n instance in the bot,
+// so the messages get translated on their way!
+bot.use(i18n.middleware());
+
+// Everything is set up now.
+// You can access translations with `t` or `translate`.
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("start-msg"));
+});
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="Deno">
+
+```ts
+import { Bot, Context } from "https://deno.land/x/grammy/mod.ts";
+import {
+  I18n,
+  I18nContextFlavor,
+} from "https://deno.land/x/grammy_i18n/mod.ts";
+
+// For TypeScript and auto-completion support,
+// extend the context with I18n's flavor:
+type MyContext = Context & I18nContextFlavor;
+
+// Create a bot as you normally would.
+// Remember to extend the context.
+const bot = new Bot<MyContext>(""); // <-- put your bot token here (https://t.me/BotFather)
+
+// Create an `I18n` instance.
+// Continue reading to find out how to configure the instance.
 const i18n = new I18n({
-  defaultLanguageOnMissing: true, // implies `allowMissing` = `true`
+  defaultLocale: "en", // see below for more information
+
+  // Load all translation files from locales/.
   directory: "locales",
-  useSession: true,
 });
 
-// You can also provide i18n data directly.
-i18n.loadLocale("en", { hey: "Hello!" });
+// Finally, register the i18n instance in the bot,
+// so the messages get translated on their way!
+bot.use(i18n);
+
+// Everything is set up now.
+// You can access translations with `t` or `translate`.
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("start-msg"));
+});
+```
+
+</CodeGroupItem>
+</CodeGroup>
 
-const bot = new Bot(process.env["BOT_TOKEN"]!);
-bot.use(session());
+`ctx.t` returns the translated message for the specified key.
+You don't have to worry about languages here.
+
+Your bot is a polyglot :tada:
```

Comment:
```suggestion
Congratulations!
Your bot is now a polyglot! :tada:
```

Reply:
I really don't like congratulating or anything like this in doc. Its a doc, not a tutorial.

Reply:
Actually, most of the time, it's both. Here, we try to explain the concepts in an understandable way, and get people up to speed. The plain API reference is elsewhere.

Can you elaborate on your reason for disliking this, beyond saying that you do?

Reply:
for me its useless, its also not helping flow but bloating the text and making visibility of important info harder.